### ISSUE TITLE
Master CI Fails

### ISSUE THREAD

**Issue by webb-ben** [geopython/pygeoapi#1445]
**Description**
Master CI/CD testing fails due to an update to starlette

**Steps to Reproduce**
Run tests with `starlette==0.34.0`

**Expected behavior**

**Screenshots/Tracebacks**
https://github.com/geopython/pygeoapi/actions/runs/7261323197/job/19782274957

**Environment**
 - OS: All
 - Python version: 3.8
 - pygeoapi version:

**Additional context**
Pinning starlette==0.33.0 fixes the problem

**Comment by ricardogsilva** [comment]
Hi @webb-ben 

I've been looking into this and, while I do think it would be generally a good idea to have pygeoapi dependencies be pinned, in this particular case I believe the better solution is for us to simply update the pygeoapi tests - and have the discussion about dependency pinning some other time :wink: 

Looking at startlette's [release notes](https://github.com/encode/starlette/releases/tag/0.34.0), the only relevant change that concerns this pygeoapi CI breakage seems to be the mention to a change to the starlette test client.

The failing pygeoapi tests are using a configuration that features:

```yaml
# snippet of pygeoapi/tests/pygeoapi-test-config-apirules.yml

server:
  url: http://localhost:5000/api

  # rest of config omitted for brevity

  api_rules:
    strict_slashes: true
    url_prefix: 'v{api_major}'
    version_header: 'X-API-Version'

```

This causes the pygeoapi tests to initialize a starlette `TestClient` by providing it a base URL of `http://localhost:5000/api`, which is the cause of the problem.

The `server.url` config value is supposed to be pygeoapi's public URL, but the pygeoapi web application is always served at the root path (_i.e._ `/`).

This means that when the public URL is modified with some additional path, as in the testing case, there must be some additional entity (usually a proxying web server) which takes care of reading the original URL (`http://localhost:5000/api/v0/` and strip away the subpath before passing it down to pygeoapi, which only understands `http://localhost:5000/v0/` (note the lack of the `/api`, sub-path).

The starlette test client, not being a proxy web server, is not able to perform such sub-path stripping. Therefore, the pygeoapi initialization of the test client should not be passing it the server's public base URL. I believe the reason why this was working before the recent change in starlette was due to this issue:

https://github.com/encode/starlette/issues/2306

Which is what motivated the starlette change.

So in conclusion, the fix for this specific problem boils down to not passing a `base_url` when initializing the startlette test client - I am able to submit a PR with it shortly.
